src/pepr/operator/controllers/keycloak/authservice/authservice.ts

```diff
@@ -79,45 +92,97 @@ export async function authservice(pkg: UDSPackage, clients: Map<string, Client>)
     K8sGateway,
     log,
   );
-  return authserviceClients;
+
+  // Return the new status objects for status update
+  return newAuthserviceClients;
 }
 
 export async function purgeAuthserviceClients(
   pkg: UDSPackage,
-  newAuthserviceClients: string[] = [],
-  isAmbient = false,
-) {
-  R.difference(pkg.status?.authserviceClients || [], newAuthserviceClients).forEach(
-    async clientId => {
-      log.info(`Removing stale authservice chain for client ${clientId}`);
-      // Use client ID directly as the waypoint ID
-      const waypointId = clientId;
-      // Get the full waypoint name for authorization policies
-      const fullWaypointName = getWaypointName(waypointId);
+  newAuthserviceClients: AuthserviceClient[] = [],
+  previousMeshMode: string,
+  currentMeshMode: string,
+): Promise<void> {
+  const isAmbient = shouldUseAmbientWaypoint(pkg);
```

Comment:
Similar comment on `isAmbient` here - I think we could skip this call and just use `currentMeshMode`?